src/main/java/edu/wpi/first/wpilibj/networktables/NetworkTableValue.java

```diff
@@ -0,0 +1,325 @@
+package edu.wpi.first.wpilibj.networktables;
+
+import java.util.Objects;
+
+/**
+ * A network table entry value.
+ */
+public class NetworkTableValue {
+  public NetworkTableValue(NetworkTableType type, long lastChange, Object value) {
```

Comment:
Do you want to leave these constructors public or only expose the static methods?

Effective Java point 1:
> Item 1: Consider static factory methods instead of constructors
> -- http://mishadoff.com/blog/effective-java/